compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -41,16 +41,19 @@ class SyntheticMethods(thisTransformer: DenotTransformer) {
 
   private var myValueSymbols: List[Symbol] = Nil
   private var myCaseSymbols: List[Symbol] = Nil
+  private var myCaseModuleSymbols: List[Symbol] = Nil
 
   private def initSymbols(implicit ctx: Context) =
     if (myValueSymbols.isEmpty) {
       myValueSymbols = List(defn.Any_hashCode, defn.Any_equals)
       myCaseSymbols = myValueSymbols ++ List(defn.Any_toString, defn.Product_canEqual,
         defn.Product_productArity, defn.Product_productPrefix, defn.Product_productElement)
+      myCaseModuleSymbols = myCaseSymbols.filter(_ ne defn.Any_equals)
```

Comment:
I would just do `myValueSymbols = defn.Any_hashCode :: (if (clazz.is(Module)) defn.Any_equals :: Nil else Nil)` instead

Reply:
That can't be done here, this is precomputing the lists for any `clazz`. We do not have access to the `clazz` in `initSymbols`